File: fulampos.cpp
#include <iostream>

// Função que soma dois valores de um array que identifica o alvo
bool somaFulampos(int notas[], int tam, int alvo) {
    int esq = 0, soma = 0;
    int dir = tam - 1;


    // o vetor já está ordenado. caso precise ordenar, posso usar a std::sort (Quick, Heap e Insertion ((sort)) O(lgN))
    while (esq <= dir) {
        soma = notas[esq] + notas[dir];

        if (soma == alvo) {
            std::cout << "Par de valores encontrado: " << notas[esq] << " e " << notas[dir] << std::endl;
            return true;
        } else if (soma < alvo) {
            // preciso de um valor um pouco maior
            esq++;
        } else { // soma > alvo | preciso de um valor um pouco menor
            dir--;
        }
    }

    std::cout << "Nenhum par encontrado que some o valor de " << alvo <<". Pode liberar o suspeito."<< std::endl;
    return false;
}

int main() {
    int carteiraSuspeito[] = {10, 20, 40, 80, 95, 110, 130};
    int tam = sizeof(carteiraSuspeito) / sizeof(carteiraSuspeito[0]); // tamanho do array
    int valorRoubado = 150;

    somaFulampos(carteiraSuspeito, tam, valorRoubado);

    return 0;
}

File: fulampos.js
// Função que soma dois valores de um array que identifica o alvo
function somaFulampos(notas, alvo) {
    let esq = 0, soma = 0;
    let dir = notas.length - 1;

    // o vetor já está ordenado. caso precise ordenar, posso usar o array.sort() O(n log n)
    while (esq <= dir) {
        soma = notas[esq] + notas[dir];

        if (soma === alvo) {
            console.log(`Par de valores encontrado: ${notas[esq]} e ${notas[dir]}`);
            return true;
        } else if (soma < alvo) {
            // preciso de um valor um pouco maior
            esq++;
        } else { // soma > alvo | preciso de um valor um pouco menor
            dir--;
        }
    }

    console.log(`Nenhum par encontrado que some o valor de ${alvo}. Pode liberar o suspeito.`);
    return false;
}


let carteiraSuspeito = [10, 20, 40, 80, 95, 110, 130];
let valorRoubado = 150;

somaFulampos(carteiraSuspeito, valorRoubado);